mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -605,6 +619,90 @@ public void onClick(View view) {
         }
     }
 
+    public static class HabpanelFragment extends Fragment implements
+            ConnectionFactory.UpdateListener {
+        Connection mConnection;
+
+        public static HabpanelFragment newInstance() {
+            return new HabpanelFragment();
+        }
+
+        @Override
+        public View onCreateView(@NonNull LayoutInflater inflater,
+                ViewGroup container, Bundle savedInstanceState) {
+            View view = inflater.inflate(R.layout.fragment_habpanel, container, false);
+
+            TextView retryButton = view.findViewById(R.id.retry_button);
+            retryButton.setOnClickListener(v -> loadHabpanel(view));
+            loadHabpanel(view);
+
+            return view;
+        }
+
+        private void loadHabpanel(View view) {
+            updateViewVisibility(false, true, view);
```

Comment:
This call seems pointless, since you'll call `updateViewVisibility` again in any case below.

Reply:
This sets the UI in the "loading" state while the connection is fetched.

Reply:
This whole method runs on the UI thread, so during the execution of this method no UI will be updated anyway (and `getUsableConnection()` executes fast anyway, as otherwise it wouldn't be OK to run on the main thread) ... so I still am pretty sure this call has no effect at all.

Reply:
Right, I remove it